vismne/visuals.py

```diff
@@ -370,13 +360,37 @@ def update_colormap(self, to_overlay=None, **kwargs):
         """
         if self._n_overlay >= 1:
             overlay = self._n_overlay - 1 if to_overlay is None else to_overlay
-            # Define the colormap data :
+            # Get limits and colormap of the overlay :
             data_lim = self._data_lim[overlay]
+            colormap = self._colormaps[overlay]
+            colormap.update(kwargs)
+            # Define the colormap data :
             col = np.linspace(data_lim[0], data_lim[1], LUT_LEN)
-            self._text2d_data[overlay, ...] = Colormap(**kwargs).to_rgba(col)
+            self._text2d_data[overlay, ...] = colormap.to_rgba(col)
             self._text2d.set_data(self._text2d_data)
             self.update()
 
+    def get_colormap(self, overlay=0):
+        """Get the colormap and limits of an overlay."""
+        colormap = self._colormaps[overlay]
+        return colormap
+
+    def add_data(self, array, smooth_mat, **kw):
```

Comment:
this is bad practice to pass kwargs. It makes the code hard to read. You don't know what arguments the kwargs contains